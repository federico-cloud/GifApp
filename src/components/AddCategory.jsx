import React, { useState } from 'react'

export const AddCategory = ({setCategories: setCategorias}) => {

    const [inputValue, setInputValue] = useState('');
    const [showError, setShowError] = useState(false);

        
    const handleInputChange = (event) => {
        setInputValue(event.target.value);
    }
    
    const handleSubmit = (event) => {
        event.preventDefault();
        
        if (inputValue.trim().length === 0){
            setShowError(true);
        } else {
            setShowError(false);
            setCategorias(categories => [inputValue, ...categories]);
            setInputValue('');
        };
    }

    return (
        <form onSubmit={handleSubmit}>
            <input 
                type="text" 
                placeholder="Search gifs of ..."
                value={inputValue}
                onChange={handleInputChange}
            />
            <div className='error-msg'>
                {
                    showError 
                    ? <span>The text musn't be empty</span>
                    : false
                }
            </div>
        </form>  
    )
}