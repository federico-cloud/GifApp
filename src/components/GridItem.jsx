import React from 'react'

export const GridItem = ({title, url}) => {
  return (
    <div className='card'>
        <a href={url} target="_blank"><img src={url} alt={`imagen de ${title}`}/></a>
    </div>
  )
}
