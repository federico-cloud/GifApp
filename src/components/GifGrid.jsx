import { useEffect, useState } from "react";
import { GridItem } from "./GridItem";
import { Box, CircularProgress } from "@mui/material";
import CloseIcon from '@mui/icons-material/Close';
import { useFetchGifs } from "../hooks/useFetchGifs";


export const GifGrid = ({category}) => {

  const {images, isLoading} = useFetchGifs(category);
  const [isClosed, setIsClosed] = useState(false);
  
  const handleClose = (event) => {
    event.target.parentNode.remove();
    setIsClosed(true);
  }

  return (
    <>
      {!isClosed && (
        <div>
          <div className="center">
            <h3>{category}</h3>
            <CloseIcon className="close" fontSize="large" onClick={handleClose} />
          </div>
          {isLoading ? (
            <Box 
              display="flex"
              justifyContent="center"
              alignItems="center"
              height="100vh"
            >
              <CircularProgress size={80} thickness={5} />
            </Box>
          ) : images.length === 0 ? (
            <p className="error-card">Images weren't found for the category {category}.</p>
          ) : (
            <div className="card-grid">
              {images.map(({ id, title, url }) => (
                <GridItem key={id} title={title} url={url} />
              ))}
            </div>
          )}
        </div>
      )}
    </>
  );
}
  
  
  
  
  
  
  
  
  