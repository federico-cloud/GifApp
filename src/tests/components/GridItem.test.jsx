import { render, screen } from "@testing-library/react"
import { GridItem } from "../../components/GridItem"

describe('Tests in GridItem ', () => {
    
    const title = 'test'
    const url   = 'https://www.google.com/';

    test('must display the componente correctly', () => { 
        const { container } = render(<GridItem title={title} url={url} />);
        
        expect(container).toMatchSnapshot();
    });

    test('must display the image with its URL and ALT', () => {
        render(<GridItem title={title} url={url} />);
        const {src, alt} = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(`imagen de ${title}`);
    })

    test('must display component title', () => {
        render(<GridItem title={title} url={url} />);
        expect(screen.getByRole('link')).toBeTruthy();
    })
});