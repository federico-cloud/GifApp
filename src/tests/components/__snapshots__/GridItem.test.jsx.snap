// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GridItem  must display the componente correctly 1`] = `
<div>
  <div
    class="card"
  >
    <a
      href="https://www.google.com/"
      target="_blank"
    >
      <img
        alt="imagen de test"
        src="https://www.google.com/"
      />
    </a>
  </div>
</div>
`;
